packages/ajax/test/AjaxClass.interceptors-cache.test.js

```diff
@@ -0,0 +1,713 @@
+    });
+
+    const actionConfig = {
+      lionCacheOptions: {
+        invalidateUrlsRegex: /foo/gi,
+      },
+    };
+
+    return ajax
+      .get('/test', actionConfig)
+      .then(() => ajax.get('/test'))
+      .then(() => {
+        expect(server.requests.length).to.equal(1);
+      })
+      .then(() => ajax.get('/foo-request-1'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .then(() => ajax.get('/foo-request-1'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .then(() => ajax.get('/foo-request-2'))
+      .then(() => {
+        expect(server.requests.length).to.equal(3);
+      })
+      .then(() => ajax.get('/foo-request-2'))
+      .then(() => {
+        expect(server.requests.length).to.equal(3);
+      })
+      .then(() => ajax.post('/test', {}, actionConfig))
+      .then(() => {
+        expect(server.requests.length).to.equal(4);
+      })
+      .then(() => ajax.get('/foo-request-1'))
+      .then(() => {
+        expect(server.requests.length).to.equal(5);
+      })
+      .then(() => ajax.get('/foo-request-2'))
+      .then(() => {
+        expect(server.requests.length).to.equal(5);
+      })
+      .finally(() => {
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it('previously cached data has to be invalidated when regex invalidation rule triggered and urls are nested', () => {
+    newCacheId();
+
+    const indexes = addCacheInterceptors(ajax, {
+      useCache: 'always',
+      timeToLive: 1000,
+    });
+
+    server.respondWith(/^\/posts$/, xhr => {
+      xhr.respond(200, { 'Content-Type': 'application/json' }, '{}');
+    });
+
+    server.respondWith(/^\/posts\/1$/, xhr => {
+      xhr.respond(200, { 'Content-Type': 'application/json' }, '{}');
+    });
+
+    const actionConfig = {
+      lionCacheOptions: {
+        invalidateUrlsRegex: /posts/gi,
+      },
+    };
+
+    return ajax
+      .get('/test', actionConfig)
+      .then(() => ajax.get('/test'))
+      .then(() => {
+        expect(server.requests.length).to.equal(1);
+      })
+      .then(() => ajax.get('/posts'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .then(() => ajax.get('/posts'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .then(() => ajax.get('/posts/1'))
+      .then(() => {
+        expect(server.requests.length).to.equal(3);
+      })
+      .then(() => ajax.get('/posts/1'))
+      .then(() => {
+        expect(server.requests.length).to.equal(3);
+      })
+      .then(() => ajax.post('/test', {}, actionConfig))
+      .then(() => {
+        expect(server.requests.length).to.equal(4);
+      })
+      .then(() => ajax.get('/posts'))
+      .then(() => {
+        expect(server.requests.length).to.equal(5);
+      })
+      .then(() => ajax.get('/posts/1'))
+      .then(() => {
+        expect(server.requests.length).to.equal(5);
```

Comment:
I would expect 6 here as well as I thought that because it's a nested url, it would not have a valid cached value for this url yet?